crates/cranelift/src/translate/code_translator.rs

```diff
@@ -2929,11 +2931,49 @@ pub fn translate_operator<FE: FuncEnvironment + ?Sized>(
             cont_type_index: _,
             tag_index: _,
             resume_table: _,
+        } => todo!("unimplemented stack switching instruction"),
+        Operator::Switch {
+            cont_type_index,
+            tag_index,
+        } => {
+            // Arguments of the continuation we are going to switch to
+            let continuation_argument_types =
+                environ.continuation_arguments(*cont_type_index).to_vec();
+            // Arity includes the continuation argument
+            let arity = continuation_argument_types.len();
+            let (contobj, switch_args) = state.peekn(arity).split_last().unwrap();
+
+            // Type of the continuation we are going to create by suspending the
+            // currently running stack
+            let created_continuation_type = continuation_argument_types.last().unwrap();
+            let created_continuation_type = created_continuation_type.unwrap_ref_type();
+
+            // Argument types of created_continuation_type. These will in turn
+            // be the types of the arguments we receive when someone switches
+            // back to this switch instruction
+            let created_continuation_arg_types = match created_continuation_type.heap_type {
+                WasmHeapType::ConcreteCont(index) => {
+                    // FIXME(frank-emrich) We need to handle the other cases.
```

Comment:
This is very ugly, but it seems like we only support module-level type indices in other places as well (most notably:  `continuation_{arguments,returns}` and `tag_{params,returns}` in `wasmtime_cranelift::func_environ`.

Reply:
It seems like we have a bunch of code that acts on type indices (i.e., something of type `EngineOrModuleTypeIndex`), but only actually support the case if it's a `ModuleInternedTypeIndex`, but not the other variants.
My new code here only supports `ModuleInternedTypeIndex` (which is what I referred to as being ugly), and the same happens in the functions `continuation_{arguments,returns}` and `tag_{params,returns}` in `wasmtime_cranelift::func_environ`.

I don't fully understand how the type indices are managed in Wasmtime, but I assume that the other variants can nevertheless pop up here (i.e.., the `RecGroup` or `Engine` variants of `EngineOrModuleTypeIndex`)?